radicle/src/test/storage.rs

```diff
@@ -55,28 +55,42 @@ impl ReadStorage for MockStorage {
         Ok(self.inventory.keys().cloned().collect::<Vec<_>>())
     }
 
-    fn repository(&self, _proj: Id) -> Result<Self::Repository, Error> {
-        Ok(MockRepository {})
+    fn repository(&self, rid: Id) -> Result<Self::Repository, Error> {
+        let doc = self
+            .inventory
+            .get(&rid)
+            .expect("Mockstorage::get: missing doc");
+        Ok(MockRepository {
+            id: rid,
+            doc: doc.clone(),
+        })
     }
 }
 
 impl WriteStorage for MockStorage {
     type RepositoryMut = MockRepository;
 
-    fn repository_mut(&self, _rid: Id) -> Result<Self::RepositoryMut, Error> {
-        Ok(MockRepository {})
+    fn repository_mut(&self, rid: Id) -> Result<Self::RepositoryMut, Error> {
+        let doc = self.inventory.get(&rid).unwrap();
+        Ok(MockRepository {
+            id: rid,
+            doc: doc.clone(),
+        })
     }
 
     fn create(&self, _rid: Id) -> Result<Self::RepositoryMut, Error> {
-        Ok(MockRepository {})
+        todo!()
```

Comment:
:eyes:

Reply:
Hahaha, ya it didn't break anything so I'm wondering what it was used for?

Reply:
Maybe not used, but the previous implementation was ok no?

Reply:
I changed `MockRepository` to have `id` and `doc` and I have no way of producing a `doc` here :thinking:

Reply:
I guess in the real world, the idea is that an `Identity` is created, giving you the `Id` to use here. But creating the repository itself is not inherently linked to the creating of the identity -- only through being given the `Id`.